File: sgf-poker/sgf-poker-ios/Core/UseCases/CreatePlayerUseCase.swift
// Core/UseCases/CreatePlayerUseCase.swift

import Foundation

/// Validates input and persists a new player.
final class CreatePlayerUseCase {

    static let maxNameLength = 50

    private let repository: PlayerRepository

    init(repository: PlayerRepository) {
        self.repository = repository
    }

    /// - Parameters:
    ///   - name: Display name for the player (trimmed internally).
    ///   - isMember: Whether the player holds membership. Defaults to false.
    ///   - isFounder: Whether the player is a founder. Defaults to false.
    /// - Returns: The newly created `Player`.
    /// - Throws: `PlayerError` describing the validation or storage failure.
    @discardableResult
    func execute(name: String, isMember: Bool = false, isFounder: Bool = false) throws -> Player {
        let trimmed = name.trimmingCharacters(in: .whitespacesAndNewlines)

        guard !trimmed.isEmpty else { throw PlayerError.emptyName }
        guard trimmed.count <= Self.maxNameLength else {
            throw PlayerError.nameTooLong(maxLength: Self.maxNameLength)
        }

        let player = Player(name: trimmed, isMember: isMember, isFounder: isFounder)
        try repository.save(player)
        return player
    }
}

File: sgf-poker-android/app/src/main/java/com/sgf/poker/usecases/player/CreatePlayerUseCase.java
package com.sgf.poker.usecases.player;

import com.sgf.poker.data.repository.PlayerRepository;
import com.sgf.poker.domain.error.PlayerException;
import com.sgf.poker.domain.model.Player;

/** Validates and persists a new player. */
public class CreatePlayerUseCase {

    public static final int MAX_NAME_LENGTH = 50;

    private final PlayerRepository repository;

    public CreatePlayerUseCase(PlayerRepository repository) {
        this.repository = repository;
    }

    public Player execute(String name, boolean isMember, boolean isFounder) {
        var trimmed = name == null ? "" : name.strip();

        if (trimmed.isEmpty()) throw PlayerException.emptyName();
        if (trimmed.length() > MAX_NAME_LENGTH) throw PlayerException.nameTooLong(MAX_NAME_LENGTH);

        var player = Player.create(trimmed, isMember, isFounder);
        repository.save(player);
        return player;
    }
}
